generator/src/main/java/com/graphicsfuzz/generator/tool/Generate.java

```diff
@@ -202,7 +212,33 @@ public static StringBuilder generateVariant(ShaderJob shaderJob,
     }
 
     if (args.getGenerateUniformBindings()) {
-      shaderJob.makeUniformBindings();
+      Optional<String> pushConstant = Optional.empty();
+      if (random.nextFloat() <= args.getPushConstantProbability()) {
+        // Get the list of (unique) uniform names in all shaders
+        final Set<String> allUniforms = new HashSet<String>();
+        for (TranslationUnit tu : shaders) {
+          if (tu.getShadingLanguageVersion().supportedPushConstants()) {
```

Comment:
Better to do this check before even testing the push constant probability.  If push constants are not supported, don't even look for opportunities to promote.

Reply:
I would thus do:

Optional<String> pushConstant = Optional.empty();
if (tu.getShadingLanguageVersion().supportedPushConstants() && random.nextFloat() <= ...) {
    // the code you have that looks through all the uniforms
}

Reply:
Done